index.bs

```diff
@@ -1739,6 +1782,219 @@ contexts that have active documents; navigation can also cause contexts to
 become inaccessible but not yet get discarded because bfcache.
 </div>
 
+## Log ## {#module-log}
+
+The <dfn export for=modules>log</dfn> module contains functionality and events
+related to logging.
+
+### Definition ### {#module-log-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+Command = (GetCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+LogEvent = (
+  LogEntryAddedEvent
+)
+</pre>
+
+### Types ### {#module-log-types}
+
+#### log.LogEntry #### {#types-log-logentry}
+
+<pre class="cddl local-cddl">
+
+LogLevel = "debug" / "info" / "warning" / "error"
+
+LogEntry = {
+  type: text,
+  level: LogLevel,
+  text: text / null,
+  timestamp: int,
+  ?stackTrace: [*StackFrame],
+  extra: { *text => any }
+}
+
+ConsoleLogExtra = {
+  type: text,
+  realm: Realm,
+  args: [*RemoteValue],
+}
+
+JavascriptLogExtra = {
+  lineNumber: int,
+  columnNumber: int,
+  url: text
+}
+
+</pre>
+
+Each log event is represented by a <code>LogEntry</code> object. This has a
+<code>type</code> property which represents the type of log entry added, a
+<code>level</code> property representing severity, a <code>text</code> property
+with the log message string itself, and a <code>timestamp</code> property
+corresponding to the time the log entry was generated. The <code>extra</code>
+property contains additional data specific to the given type.
```

Comment:
What is the `type` and is it different from the `level`? In [CDP there is only `type` in the meaning of `level`](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#event-consoleAPICalled).

Reply:
The API here covers both `Runtime.ConsoleAPICalled` and `Log.EntryAdded` events from CDP. So the `type` tells you about the kind of log entry it is (e.g. from a `console` method call in js, an unhandled js exception, or something else), `level` is a coarse levelling to allow filtering across all types, and the type-specific information is in `extra`. In particular to get the particular console method call you need to look in `extra`, not `level`. Does that make sense?

Reply:
I got it, thanks.